tea.go

```diff
@@ -384,6 +384,14 @@ func (p *Program) eventLoop(model Model, cmds chan Cmd) (Model, error) {
 							continue
 						}
 
+						// process nested sequence messages in order
+						if sequenceMsg, ok := msg.(sequenceMsg); ok {
+							for _, cmd := range sequenceMsg {
+								p.Send(cmd())
```

Comment:
What if we have deeply nested sequences (3+)? Imagine we have a `cmd` like:
```
sequenceMsg: [
    sequenceMsg: [
        sequenceMsg: [customMsg11, customMsg12],
        sequenceMsg: [customMsg21, customMsg22]
    ]
]
```
where each `Msg` represents a `cmd` and what will be returned after running it.

Doesn't this solution only work for the first two sequences? The outermost sequence is handled by line 363, and then the second sequence is handled here, but the third and fourth sequences (with the `customMsg*`s) will be done concurrently -- `customMsg11` will happen before `customMsg12`, but `customMsg21` could occur before `customMsg11`, after `customMsg11`, or after `customMsg12`. Am I missing something?